Plugin.cs

```diff
@@ -28,14 +30,59 @@ public class Plugin : IViewer
         private bool _isDirectory;
 
         public int Priority => -5;
-
+        private static readonly int FILES_MAX = 300;
         public void Init()
         {
         }
 
+        /// <summary>
+        /// A safe way to get all the files in a directory and sub directory without crashing on UnauthorizedException or PathTooLongException
+        /// </summary>
+        /// <param name="rootPath">Starting directory</param>
+        /// <param name="patternMatch">Filename pattern match</param>
+        /// <param name="searchOption">Search subdirectories or only top level directory for files</param>
+        /// <returns>List of files</returns>
+        public static IEnumerable<string> GetDirectoryFiles(string rootPath, string patternMatch, SearchOption searchOption)
+        {
+            var foundFiles = Enumerable.Empty<string>();
+
+            if (searchOption == SearchOption.AllDirectories)
+            {
+                try
+                {
+                    IEnumerable<string> subDirs = Directory.EnumerateDirectories(rootPath);
+                    foreach (string dir in subDirs)
+                    {
+                        if (foundFiles.Count() < FILES_MAX)
```

Comment:
Could you `break` (or even `return`) if the condition fails, so that each recursive call can travel up as soon as it exceeds `FILE_MAX`?

Reply:
Less than FILES_MAX, just return the file count of folders. Use try, catch to catch UnauthorizedException and so on.

Reply:
Sorry, I did not get it. 

What I am trying to say is, if the condition fails, and the count exceeds, add an else block to either break or return so that we stop additional looping.